tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,3176 @@
+
+  // Evaluate analytic solution
+  const auto vars =
+      solution.variables(x, t, typename Solution::template tags<DataVector>{});
+  const auto& lapse = get<gr::Tags::Lapse<>>(vars);
+  const auto& shift = get<gr::Tags::Shift<SpatialDim>>(vars);
+  const auto& spatial_metric = get<gr::Tags::SpatialMetric<SpatialDim>>(vars);
+
+  // Get ingredients
+  const auto& spacetime_metric =
+      gr::spacetime_metric(lapse, shift, spatial_metric);
+  const auto& spacetime_unit_normal_one_form =
+      gr::spacetime_normal_one_form<SpatialDim, Frame::Inertial, DataVector>(
+          lapse);
+  const auto& det_spatial_metric =
+      determinant_and_inverse(spatial_metric).first;
+  Scalar<DataVector> sqrt_det_spatial_metric(sqrt(get(det_spatial_metric)));
+
+  // Initialize settings
+  const double amp_coef_L2 = pdist(generator);
+  const int exp_L2 = idist(generator);
+  const double t_start_L2 = pdist(generator) * 0.1;
+  const double sigma_t_L2 = pdist(generator) * 0.2;
+  const double r_max = pdist(generator) * 0.7;
+  const double roll_on_L2 =
+      GeneralizedHarmonic::DampedHarmonicGauge_detail::roll_on_function(
+          t, t_start_L2, sigma_t_L2);
+  const auto& weight =
+      GeneralizedHarmonic::DampedHarmonicGauge_detail::weight_function<
+          SpatialDim, Frame::Inertial, DataVector>(x, r_max);
+
+  const auto gauge_h_init = make_with_value<typename db::item_type<
+      GeneralizedHarmonic::Tags::InitialGaugeH<SpatialDim, Frame::Inertial>>>(
+      x, 0.);
+  typename db::item_type<
+      GeneralizedHarmonic::Tags::GaugeH<SpatialDim, Frame::Inertial>>
+      gauge_h{};
+  GeneralizedHarmonic::damped_harmonic_h<SpatialDim, Frame::Inertial>(
+      make_not_null(&gauge_h), gauge_h_init, lapse, shift,
+      sqrt_det_spatial_metric, spacetime_metric, t, x, 0., amp_coef_L2, 0., 0,
+      exp_L2, 0, /* exponents */
+      // roll on function parameters for lapse / shift terms
+      -1., -1., -1., -1., t_start_L2, sigma_t_L2, -1., -1.,
+      r_max /* weight function */);
+
+  /*
+   * Evaluate analytic solution for Schwarzschild, ab initio.
+   */
+  const double mass = solution.mass();
+  const std::array<double, SpatialDim> center = solution.center();
+  // 1) shift coords with BH at the center
+  auto x_minus_center = x;
+  // DataVector r_squared(get_size(x), 0.);
+  auto r_squared = get<0>(x);
+  r_squared -= get<0>(x);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    x_minus_center.get(i) -= gsl::at(center, i);
+    r_squared += square(x_minus_center.get(i));
+  }
+  const DataVector r_ = sqrt(r_squared);
+  // 2) compute scalar function that defines KerrSchild coords
+  const DataVector H = mass * (1. / r_);
+  const DataVector H2 = H * 2.;
+  // 3) compute lapse
+  const Scalar<DataVector> lapse_ab_initio{sqrt(1. / (1. + H2))};
+
+  const auto& spacetime_unit_normal_one_form_ab_initio =
+      gr::spacetime_normal_one_form<SpatialDim, Frame::Inertial, DataVector>(
+          lapse_ab_initio);
+  /* *** */
```

Comment:
I'll just leave it to you to remove all of these weird comments

Reply:
removed